service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -99,64 +96,166 @@ public void setRestTemplate(RestTemplate restTemplate) {
                 }
             } else {
-                return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_SERVICE_ERROR),
-                        HttpStatus.SERVICE_UNAVAILABLE);
-            }
-        } catch (REMGenerateException e1) {
-            if (e1.getMessage().contains(Integer.toString(HttpStatus.NOT_ACCEPTABLE.value()))) {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.NOT_ACCEPTABLE);
-            }
-            else if (e1.getMessage().contains(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.EXPECTATION_FAILED);
+                return createResponseEntity(RemremGenerateServiceConstants.JSON_STATUS_CODE, RemremGenerateServiceConstants.JSON_STATUS_RESULT, RemremGenerateServiceConstants.JSON_ERROR_MESSAGE_FIELD,
+                        HttpStatus.BAD_REQUEST, "Invalid JSON format,expected either single template or array of templates", "fail", errorResponse, HttpStatus.BAD_REQUEST.value());
             }
-            else if (e1.getMessage().contains(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.EXPECTATION_FAILED);
-            }
-            else if (e1.getMessage()
-                       .contains(Integer.toString(HttpStatus.SERVICE_UNAVAILABLE.value()))) {
-                return new ResponseEntity<>(parser.parse(RemremGenerateServiceConstants.NO_ER),
-                        HttpStatus.SERVICE_UNAVAILABLE);
+        } catch (REMGenerateException | JsonSyntaxException e) {
+            return handleException(e);
+        }
+    }
+
+    public ResponseEntity<JsonObject> createResponseEntity(String statusCode, String statusResult, String statusMessage,
+                                                           HttpStatus status, String errorMessage, String resultMessage, JsonObject errorResponse,
+                                                           int statusCodeValue) {
+        errorResponse.addProperty(statusCode, statusCodeValue);
+        errorResponse.addProperty(statusResult, resultMessage);
+        errorResponse.addProperty(statusMessage, errorMessage);
+        return new ResponseEntity<>(errorResponse, status);
+
+    }
+
+    private ResponseEntity<JsonObject> handleException(Exception e) {
+        JsonObject errorResponse = new JsonObject();
+        String exceptionMessage = e.getMessage();
+        if (e instanceof REMGenerateException) {
+            List<HttpStatus> statuses = List.of(
+                    HttpStatus.NOT_ACCEPTABLE, HttpStatus.EXPECTATION_FAILED, HttpStatus.SERVICE_UNAVAILABLE, HttpStatus.UNPROCESSABLE_ENTITY
+            );
+            for (HttpStatus status : statuses) {
+                if (exceptionMessage.contains(Integer.toString(status.value()))) {
```

Comment:
I guess there should be more strict test, not just test for occurrence in message. The message is a JSON data.